### DISCUSSION TITLE
Automatic Connection to Banks via Plaid or other?

### DISCUSSION THREAD

**Discussion by sameleff** [ghostfolio/ghostfolio#2608] [Q&A] [answered]
I see this has been mentioned [elsewhere ](https://github.com/ghostfolio/ghostfolio/issues/782#issuecomment-1086296670)- but is it currently possible to connect to financial institutions directly in order to download transactions and import them into Ghostfolio via Plaid or some other API service such as GoCardless? If not - what is required in order to progress the Ghostfolio API to enable this feature?

**Comment by dtslvr** [comment, marked as answer]
Hi @sameleff

You could write a script (preferably in _TypeScript_) that connects to the data connector of your choice, fetches the data, transforms it  to the Ghostfolio format and imports it via the [Ghostfolio API](https://github.com/ghostfolio/ghostfolio#import-activities).

At the moment we don't include such specific logic in this project. However, if you publish your script as a [community project](https://github.com/topics/ghostfolio), others will also benefit and there is a possibility that it will be integrated into Ghostfolio at some point.

**Comment by gmag11** [comment]
It would be nice to develop a general connector so that specific connectors for different brokers or exchanges could be integrated.

**Comment by backtest-graveyard** [comment]
For anyone who lands here from the links in later threads: I took the route @dtslvr described above, a separate script that pulls from a data connector and imports via the Ghostfolio API. It uses SnapTrade, runs daily, read-only, and covers the brokers on their list (Trading 212, DEGIRO, IBKR, Schwab, Fidelity, Robinhood and more).

It's free and open source: https://github.com/backtest-graveyard/brokersync

It syncs current holdings and cost basis, not the full trade history. @gmag11, it's close to the general-connector idea you described.
